DirectProgramming/DPC++FPGA/Tutorials/Features/latency_control/src/latency_control.cpp

```diff
@@ -0,0 +1,123 @@
+//==============================================================
+// Copyright Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+// =============================================================
+#include <CL/sycl.hpp>
+#include <numeric>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+
+// dpc_common.hpp can be found in the dev-utilities include folder.
+// e.g., $ONEAPI_ROOT/dev-utilities/include/dpc_common.hpp
+#include "dpc_common.hpp"
+
+using namespace sycl;
+
+using BurstCoalescedLSU = ext::intel::experimental::lsu<
+    ext::intel::experimental::burst_coalesce<true>,
+    ext::intel::experimental::statically_coalesce<false>>;
+
+constexpr size_t kNum = 5;
+
+int Operation(int a) { return a * 3 + 2; } // Arbitrary operations.
+
+// Forward declare the kernel names in the global scope.
+// This FPGA best practice reduces name mangling in the optimization reports.
+class LatencyControl;
+
+// Runs the Kernel.
+void KernelRun(const std::vector<int> &in_data, std::vector<int> &out_data,
+               const size_t &size) {
+#if defined(FPGA_EMULATOR)
+  ext::intel::fpga_emulator_selector device_selector;
+#else
+  ext::intel::fpga_selector device_selector;
+#endif
+
+  try {
+    // Create the SYCL device queue.
+    queue q(device_selector, dpc_common::exception_handler,
+            property::queue::enable_profiling{});
+
+    buffer in_buffer(in_data);
+    buffer out_buffer(out_data);
+
+    q.submit([&](handler &h) {
+      accessor in_accessor(in_buffer, h, read_only);
+      accessor out_accessor(out_buffer, h, write_only, no_init);
+
+      h.single_task<LatencyControl>([=]() [[intel::kernel_args_restrict]] {
+        auto in_ptr = in_accessor.get_pointer();
+        auto out_ptr = out_accessor.get_pointer();
+
+        for (size_t i = 0; i < size; i++) {
+          int value = BurstCoalescedLSU::load(
+              in_ptr + i, ext::oneapi::experimental::properties(
+                              ext::intel::experimental::latency_anchor_id<0>));
+
+          // Constrain the latency between the above load and the below store.
```

Comment:
This comment seems not to be in the correct place

Reply:
It seems to be commenting the "Operation" rather than the actual constraint which is located below.
Would that make sense to put it next to the constraint?